pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,191 @@
+package describe
+
+import (
+	"fmt"
+	"strings"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	utilerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+	"github.com/golang/glog"
+	"github.com/gonum/graph"
+	"github.com/gonum/graph/encoding/dot"
+	"github.com/gonum/graph/path"
+	"github.com/gonum/graph/traverse"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	buildedges "github.com/openshift/origin/pkg/build/graph"
+	buildgraph "github.com/openshift/origin/pkg/build/graph/nodes"
+	"github.com/openshift/origin/pkg/client"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+	"github.com/openshift/origin/pkg/util/parallel"
+)
+
+// NotFoundErr is returned when the imageStreamTag (ist) of interest cannot
+// be found in the graph. This doesn't mean though that the ist does not
+// exist. A user may have an image stream without a build configuration
+// pointing at it. In that case, the ist of interest simply doesn't have
+// other dependant ists
+type NotFoundErr string
+
+func (e NotFoundErr) Error() string {
+	return fmt.Sprintf("couldn't find image stream tag: %q", string(e))
+}
+
+// ChainDescriber generates extended information about a chain of
+// dependencies of an image stream
+type ChainDescriber struct {
+	client.Interface
```

Comment:
Why should this also act as a `client.Interface`?  I don't think this should be anonymous.  Also, can you choose the specific interfaces you need?  `BuildConfigNamespacer`?